lib/promise.js

```diff
@@ -19,6 +19,12 @@ export function delay(ms) {
   });
 }
 
+export function delayThenReject(ms) {
+  return new Promise((resolve, reject) => {
```

Comment:
await is an operation on a promise and async denotes that a function returns a promise and uses await in its implementation.

In this case, we need to wrap setTimeout, which is not a promise, so we need to make one.

In general, we should be preferring async/await in modern code, because it's less error-prone (and usually easier to read) than Promise.then chaining, especially in the face of loops or proper exception handling.

Reply:
Cool, thanks for the explanation.